tests/testdata/provider/testdata_pg.sh

```diff
@@ -5,10 +5,10 @@ DB=${1:-qgis_test}
 cd "$(dirname "$0")"/../../../ || exit 1
 
 SCRIPTS="
+  tests/testdata/provider/testdata_pg_role.sql
```

Comment:
If the order matters here, a comment would be good, unless someone decides to sort them alphabetically.

Reply:
Good idea, implemented and rebased/force-pushed - thanks for looking !

Reply:
That's a.. pretty obscure comment. I would have gone for something like:

# create the role first, this is super important
"tests/testdata/provider/testdata_pg_role.sql"

# create the database and other stuff
"tests/testdata/provider/testdata_pg.sql"
# ...

# create the foreign keys at the end
"tests/testdata/provider/testdata_pg_relations.sql"

I'm sure there's a way to intersperse comments and strings (maybe using an array), but I'm not going to bother looking it up. Maybe "order matters" is enough for the reader.